### PR TITLE
Travis CI integration

### LINKED ISSUE #13533: Suggestion: integrate with CI and require PRs to pass
Currently, it's difficult to contribute, when tests are already broken and lint does not pass. It would also make maintenance easier as everyone would have immediate feedback if the PR can be merged.

There have been discussions in the past, but they invariably end with "some code is not supposed to pass lint/tests". It would be nice to revisit this decision as three.js has no measures in place to tell a potential contributor if his code is remotely valid.

Thanks for considering!

### PR DESCRIPTION
**Pull request by dbkaplun**
Closes #13533 

Tests and lint pass on dev branch, making CI integration possible. This PR integrates Travis CI for THREE.js. Travis is a continuous integration provider that is completely free for open source projects.

The [Travis build passes for this branch](https://travis-ci.org/dbkaplun/three.js/builds/359888191), on my fork.

Once this PR is merged, an owner of the mrdoob/three.js repo will need to enable Travis by following steps 1 and 2 [here](https://docs.travis-ci.com/user/getting-started/#To-get-started-with-Travis-CI). This will allow Github to automatically show status of tests (pass/fail) for new PRs. (This can be done in advance without affecting the repository.)

Thanks.

### CODE REVIEW

File: src/math/Math.js [hunk 1]
```diff
@@ -31,7 +31,7 @@ var _Math = {
 				lut[ d2 & 0x3f | 0x80 ] + lut[ d2 >> 8 & 0xff ] + '-' + lut[ d2 >> 16 & 0xff ] + lut[ d2 >> 24 & 0xff ] +
 				lut[ d3 & 0xff ] + lut[ d3 >> 8 & 0xff ] + lut[ d3 >> 16 & 0xff ] + lut[ d3 >> 24 & 0xff ];
 
-			// .toUpperCase() here flattens concatenated strings to save heap memory space. 
+			// .toUpperCase() here flattens concatenated strings to save heap memory space.
```
Comment (Mugen87) [inline src/math/Math.js]: Maybe you can remove this unrelated change from the PR?

Comment (dbkaplun) [reply src/math/Math.js]: @Mugen87 this change allows linting to pass.

Comment (Mugen87) [reply src/math/Math.js]: Oh, then it's of course fine :blush:

**Review by Mugen87 — APPROVED** [review]

**Comment by mrdoob** [comment]
Okay, lets try it... 🙂

**Comment by mrdoob** [comment]
Thanks!

**Comment by Mugen87** [comment]
@dbkaplun Isn't it also possible to add some kind of Travis badge or image to the project's README which visualizes the build status? I've seen this sometimes at github and it's kind of cool :blush:

**Comment by looeee** [comment]
@Mugen87 like this? https://docs.travis-ci.com/user/status-images/

**Comment by Mugen87** [comment]
Yeah, exactly.

**Comment by hexagon6** [comment]
@Mugen87 you can add this Markdown to the README.md: `[Build Status](https://travis-ci.org/mrdoob/three.js)` -> [Build Status](https://travis-ci.org/mrdoob/three.js)

**Comment by hexagon6** [comment]
And adjust `branch=${branch}` accordingly

**Comment by mrdoob** [comment]
Is it possible to run travis only when files in `/src` have been changed?
No need to run it when files in `/docs`, `/examples`, ... are changed.

**Pull request merged**